CDCgov/prime-reportstream
# 14350: update submission to be agnostic of steps

This PR updates the submission history endpoint to be agnostic of the underlying action names

Test Steps:
On master
1. Send a submission
2. Get the report Id and execute the submission endpoint
```
curl --location 'http://localhost:7071/api/waters/report/{REPORT_ID}/history' \
--header 'authentication-type: okta' \
--header 'Authorization: Bearer dummy'
```

Do the same on this branch and confirm the outputs are the same (they might differ in order FYI)

## Changes
- Updates how the submission history is generated
- Adds the intellij project settings
- Adds DB tests for the submission history endpoints

## Checklist

### Testing

### Process

## Linked Issues
#14350 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
